### ISSUE TITLE
Roundcube. Connection to storage server failed

### ISSUE THREAD

**Issue by mindaugasdi** [hestiacp/hestiacp#1119]
### In a few words, please describe the issue that you're experiencing:
Trying to login to mail, Roundcube throws error "Connection to storage server failed".

### What steps did you take when the issue occured?
1. Install hestia (fresh install).
2. Add user, domain, mail domain, mail address. 
3. Go to address I specified in settings "Webmail Alias", it opens Roundcube.
4. Trying to login, it does not work.

### Expected behavior:
Logs in successfully or displaying some other error like password incorrect.

### Operating system:
Ubuntu 18.04.5 LTS

### Hestia Control Panel version:
v1.2.3

### Additional notes:

What I tried:
1. Checked if dovecot is running.
  - It is running in HESTIA/list/server/, but memory used 0.
  - Terminal HTOP displays 5 dovecot processes running.
2. Tried restarting dovecot in hestia, and in terminal.
3. Dovecot log displays no info at the time of the error. Only info at other times. Mostly Pop3, imap spam logins, server restart msgs.
4. Went through existing issues, but those are old.
5. Roundcube log has these errors `[27-Aug-2020 11:34:03 +0000]: <8a938642> IMAP Error: Login failed for ........ from ....(X-Forwarded-For: ....). Could not connect to localhost:143: Unknown reason in /usr/share/roundcube/program/lib/Roundcube/rcube_imap.php on line 196 (POST /?_task=login&_action=login)`

**Comment by jaapmarcus** [comment]
Try to login with an Imap client to see that is working (Thunderbird for example)

**Comment by mindaugasdi** [comment]
Tested thunderbird and it works - I can receive and send mail.

There is other (related) bug - opening roundcube from hestia directly does not work, UI link points to non existing default alias (webmail.*), even though it is changed to non-default one in settings. In other context, this settings works, for example, it generated SSL certificate well and domain opens.

**Comment by jaapmarcus** [comment]
Roundcube is only allowed to run on a subdomain. Normally when use of DNS this subdomain is created how ever when running a DNS server by a different provider Hestia is not able to create them. (Cloudflare or a different provider) 

Create webmail subdomain and create ssl certificates and try again.

**Comment by mindaugasdi** [comment]
I use subdomain already, but non-default one (translated to my language).
~~DNS A record is ' * ' pointing all subdomains to the server.~~

I will try:
~~1. Explicitly create subdomain A record for this, but I don't get the reason behind this.~~
2. Will try to use default subdomain 'webmail'.

And will post what works.

**Comment by jaapmarcus** [comment]
https://bobcares.com/blog/roundcube-connection-to-storage-server-failed/

As Imap is available don't think it is a issue with Dovecote..

`sudo systemctl status dovecot` could give some additional information

**Comment by ScIT-Raphael** [comment]
Any update @mindaugasdi? Just using let's encrypt for mail domain and webmail.domain.tld as url and don't have any troubles.

**Comment by mindaugasdi** [comment]
I did try webmail subdomain, but it did not work. Will write in more detail tomorrow. Thanks for keeping an eye on this.

**Comment by mindaugasdi** [comment]
Updated issue description with this error in roundcube log:
`[27-Aug-2020 11:34:03 +0000]: <8a938642> IMAP Error: Login failed for ........ from ....(X-Forwarded-For: ....). Could not connect to localhost:143: Unknown reason in /usr/share/roundcube/program/lib/Roundcube/rcube_imap.php on line 196 (POST /?_task=login&_action=login)`

**Comment by ScIT-Raphael** [comment]
@mindaugasdi could you share with which components you installed hestia aswell? Just did a fresh installation on ubuntu 20.04 and I'm not able to reproduce the issue.

According to your log, it looks like there is a connection issue to imap port. Are you sure, that dovecot service is up and running aswell as the port 143 is in use? What happens if you run a `telnet mail.domain.tld 143`? You should get a output like: `* OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE START TLS AUTH=PLAIN AUTH=LOGIN] Mail Delivery Agent`

**Comment by mindaugasdi** [comment]
Installated with:
```
wget https://raw.githubusercontent.com/hestiacp/hestiacp/release/install/hst-install.sh
bash hst-install.sh --multiphp yes --api no 
```

Telnet outputs:
`* OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE STARTTLS AUTH=PLAIN AUTH=LOGIN] Mail Delivery Agent`

Tested again and IMAP is working, I can send and receive messages using Thunderbird. Dovecot is running. And roundcube cannot login and throws that error, even though it has lets encrypt SSL and subdomain is webmail.domain.tld

**Comment by mindaugasdi** [comment]
I remembered what I changed additionally, it may very well be the problem. I will check and write back.

**Comment by ScIT-Raphael** [comment]
Infact that we arent able to reproduce the issue and it sounds like you manualy adjusted something, I gonna close this issue report. Please feel free to update it with the change you did, just in case you need additional informations.

**Comment by mindaugasdi** [comment]
Cause of the problem:
* I disabled some php functions following forums security recommendations. Because 'fsockopen' function was disabled, Roundcube was not able to connect. Its not a good idea to disable functions and follow similar recommendations, because many many things kept failing in the past.

Proposal:
* Usually PHP complains about functions being disabled, but its unclear where to look for roundcube PHP logs. There these logs can be found? Central PHP logs do not log it. Would it be possible to make these logs discoverable for the user?

I changed back the webmail alias to a different one and it works, so whatever alias works well.

Sorry for the trouble. And thanks for help.
